core/src/main/scala/com/gu/tableversions/core/TableVersions.scala

```diff
@@ -13,42 +17,96 @@ trait TableVersions[F[_]] {
     * Start tracking version information for given table.
     * This must be called before any other operations can be performed on this table.
     */
-  def init(table: TableName, isSnapshot: Boolean, userId: UserId, message: UpdateMessage, timestamp: Instant): F[Unit]
+  def init(table: TableName, isSnapshot: Boolean, userId: UserId, message: UpdateMessage, timestamp: Instant): F[Unit] =
+    handleInit(table) {
+      val initialUpdate = TableUpdate(userId, message, timestamp, operations = List(InitTable(table, isSnapshot)))
+      TableState(currentVersion = initialUpdate.metadata.id, updates = List(initialUpdate))
+    }
 
-  /** Get details about partition versions in a table. */
-  def currentVersion(table: TableName): F[TableVersion]
+  /**
+    * Get details about partition versions in a table.
+    */
+  def currentVersion(table: TableName)(implicit F: Sync[F]): F[TableVersion] =
+    // Derive current version of a table by folding over the history of changes
+    // until either the latest or version marked as 'current' is reached.
+    for {
+      ts <- tableState(table)
+      matchingUpdates = ts.updates.span(_.metadata.id != ts.currentVersion)
+      updatesForCurrentVersion = matchingUpdates._1 ++ matchingUpdates._2.take(1)
```

Comment:
would adding separate descriptive names to matching updates make this clearer?
something like this (but with better names probably) :
```scala
      (nonCurrent, current) = ts.updates.span(_.metadata.id != ts.currentVersion)
      updatesForCurrentVersion = nonCurrent++ current.take(1)
````

Reply:
Yes indeed! I thought this wasn't actually possible due to the but I guess this is OK because it's inside a `map` not a `flatMap`. Well spotted!